android/app/src/main/kotlin/coop/polypoly/polypod/PodApi.kt

```diff
@@ -134,7 +134,7 @@ open class PodApi(
         var path = ""
         if (!args[0].isNilValue) { path = args[0].asStringValue().toString() }
         val result = polyOut.readdir(path)
-        return ValueFactory.newArray(result.map { ValueFactory.newString(it) })
+        return ValueFactory.newArray(result.map { ValueFactory.newString(it["id"]) })
```

Comment:
Seems like it's returning the same it returned before, an array of IDs (sometimes also called URLs) -> Why not `{id: ..., path: ...}` here?

Reply:
Yess! that is the issue I have been tackling and after examining it with JJ we understood that the function being called here is not the function that is being called in browserPod but rather `fs.readdir()` which is in default.js, in which case it will continue returning exactly what it was returning before independently from any changes in browserPod! So we have chosen to make a mock version of the `readdir `function in browserPod and put it inside default.ts which is ultimately the `readdir `function that is being called by the test

Reply:
The function being called here is your implementation in Kotlin, `polyOut.readdir`, but you're disregarding the result instead of returning it properly. You need to use `ValueFactory` to make an array of _objects_, not of strings, otherwise calling `polyOut.readdir` from a feature running in the Android polyPod would not return the expected result (an array of objects with `id` and `path`), but rather an array of strings, like before.

What you're referring to sounds like the test failure - fair enough, monkey patching MemFS with a `readdir` function that satisfies the new interface sounds like on way to get them green again :D But disabling the tests that cause problems would be fine too, they are merely testing unused code anyway. (I wouldn't quite _remove_ them yet, because we might still want to use them against the `BrowserPod` implementation later.)